### PR TITLE
Adds documentation for new AWS CloudWatch sinks

### PR DESCRIPTION
**Pull request by TimSimpson**

### CODE REVIEW

File: docs/cloud/integrations/README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Integrations
 
-Event Store Cloud offers integrations between internal sources such as cluster health, [issue](#issues) detection, [notifications](#notifications) events and sinks which include external services such as Slack and OpsGenie.
+Event Store Cloud offers integrations between internal sources such as cluster health, [issue](#issues) detection, [notifications](#notifications) events, EventStore DB [logs], EventStore DB [metrics] and sinks which include external services such as Slack and Aws CloudWatch.
```
Comment (ylorph) [inline docs/cloud/integrations/README.md]:
```suggestion
Event Store Cloud offers integrations between internal sources such as cluster health, [issue](#issues) detection, [notifications](#notifications) events, EventStore DB [logs](#logs), EventStore DB [metrics](#metrics) and sinks which include external services such as Slack and Amazon CloudWatch.
```

File: docs/cloud/integrations/README.md [hunk 2]
```diff
@@ -10,8 +10,12 @@ Currently, supported sources include:
 
 * [Issues](#issues) - issues represent a potentially problematic condition detected inside a cluster or other Event Store Cloud resource. Issues consist of multiple "open" events and a single "closed" event and have different levels of severity.
 
-* [Notifications](#notifications) - notifications are noteworthy events detected within Event Store Cloud resources or the backend. For example notifications may be emitted when a cluster fails to provision.
+* [Logs](#logs) - these are the logs generated by EventStore DB itself.
 
+* [Metrics](#metrics) - these are the metrics generated by both the EventStore Cloud issue detection processes running on the each cluster node and EventStore DB itself.
```
Comment (ylorph) [inline docs/cloud/integrations/README.md]:
```suggestion
* [Metrics](#metrics) - these are the metrics generated by both EventStore Cloud issue detection processes running on each cluster node and EventStore DB itself.
```

File: docs/cloud/integrations/README.md [hunk 3]
```diff
@@ -60,19 +64,231 @@ If, for some reason, the instances backing a cluster fail to provision the resou
 
 If the volume fails to expand while expanding an instances size a notification event is created with the message `Cluster volumes failed to provision`.
 
+### Logs
+
+**NOTE: Logs are currently in beta and may change over time. The data coming from this source may change over time.**
```
Comment (ylorph) [inline docs/cloud/integrations/README.md]:
```suggestion
::: note
Logs are currently in beta and may change over time. The data coming from this source may change over time.**
:::
```

File: docs/cloud/integrations/README.md [hunk 4]
```diff
@@ -60,19 +64,231 @@ If, for some reason, the instances backing a cluster fail to provision the resou
...
+
+Logs are sent in the form of JSON objects. Each object's "message" field contains the original structured event from EventStore DB itself.
+
+### Metrics
+
+**NOTE: Logs are currently in beta and may change over time. The data coming from this source may change over time.**
```
Comment (ylorph) [inline docs/cloud/integrations/README.md]:
```suggestion
::: note
Logs are currently in beta and may change over time. The data coming from this source may change over time.**
:::
```

File: docs/cloud/integrations/README.md [hunk 5]
```diff
@@ -60,19 +64,231 @@ If, for some reason, the instances backing a cluster fail to provision the resou
...
+
+Metrics are name / value pairs which are generated by EventStore DB itself as well as detection processes running on the host.
```
Comment (ylorph) [inline docs/cloud/integrations/README.md]:
```suggestion
Metrics are name / value pairs generated by EventStore DB itself as well as detection processes running on the host.
```

File: docs/cloud/integrations/README.md [hunk 6]
```diff
@@ -60,19 +64,231 @@ If, for some reason, the instances backing a cluster fail to provision the resou
...
+
+
 ## Integration sinks
 
 "Sinks" are services outside the Event Store Cloud which events from sources can be forwarded to.
 
+* [AwsCloudWatchLogs](#aws-cloudwatch-logs-sink) - **BETA** AWS CloudWatch allows you to track metrics, display them with create custom dashboards, create alarms from them, etc.
```
Comment (ylorph) [inline docs/cloud/integrations/README.md]:
```suggestion
* [AwsCloudWatchLogs](#aws-cloudwatch-logs-sink) - **BETA** Amazon CloudWatch allows you to track metrics, display them with custom dashboards, create alarms from them.
```

File: docs/cloud/integrations/README.md [hunk 7]
```diff
@@ -60,19 +64,231 @@ If, for some reason, the instances backing a cluster fail to provision the resou
...
+
+* [AwsCloudWatchMetrics](#aws-cloudwatch-metrics-sink) - **BETA** AWS CloudWatch allows for logs to be uploaded, viewed and searched, and consumed by other AWS services.
```
Comment (ylorph) [inline docs/cloud/integrations/README.md]:
```suggestion
* [AwsCloudWatchMetrics](#aws-cloudwatch-metrics-sink) - **BETA** Amazon CloudWatch allows for logs to be uploaded, viewed and searched, and consumed by other AWS services.
```

File: docs/cloud/integrations/README.md [hunk 8]
```diff
@@ -60,19 +64,231 @@ If, for some reason, the instances backing a cluster fail to provision the resou
...
+
 * [OpsGenie](#opsgenie-sink) - OpsGenie is an alerting and incidence response tool. It is possible to set up integrations to create OpsGenie alerts when cluster health issues are detected.
 
 * [Slack](#slack-sink) - Slack is a communication platform. It is possible to set up integrations which send Slack messages when issues and notifications are created or updated.
 
+### Aws CloudWatch Logs sink
+
+**NOTE: The AwsCloudWatchLogs sink is currently in closed beta and may change over time. To access it, please contact support.**
```
Comment (ylorph) [inline docs/cloud/integrations/README.md]:
```suggestion
::: note
The AwsCloudWatchLogs sink is currently in closed beta and may change over time. To access it, please contact support.
:::
```

File: docs/cloud/integrations/README.md [hunk 9]
```diff
@@ -60,19 +64,231 @@ If, for some reason, the instances backing a cluster fail to provision the resou
...
+
+AWS CloudWatch is a monitoring service. The "logs" portion of its functionality is separated into it's own integration in the EventStore Cloud for ease of use.
```
Comment (ylorph) [inline docs/cloud/integrations/README.md]:
```suggestion
AWS CloudWatch is a monitoring service. The "logs" portion of its functionality is separated into it's own integration in EventStore Cloud for ease of use.
```

File: docs/cloud/integrations/README.md [hunk 10]
```diff
@@ -60,19 +64,231 @@ If, for some reason, the instances backing a cluster fail to provision the resou
...
+
+You can cause all of the logs generated by each EventStore DB running in your clusters by creating a custom AWS CloudWatch log group, an IAM user, and setting up an integration as described below.
+
+*NOTE: If you're using Terraform, instructions on how to perform the steps below can be found [here](https://github.com/EventStore/terraform-provider-eventstorecloud/blob/trunk/docs/resources/integration_awscloudwatch_logs.md#example-usage).*
```
Comment (ylorph) [inline docs/cloud/integrations/README.md]:
```suggestion
:::note 
If you're using Terraform, instructions on how to perform the steps below can be found [here](https://github.com/EventStore/terraform-provider-eventstorecloud/blob/trunk/docs/resources/integration_awscloudwatch_logs.md#example-usage).
:::
```

File: docs/cloud/integrations/README.md [hunk 11]
```diff
@@ -60,19 +64,231 @@ If, for some reason, the instances backing a cluster fail to provision the resou
...
+
+Enter a log group. If you have multiple projects it could be a good idea to include it in the name, such as `EventStoreLogs-Production`.
+
+Select a retention policy depending on your needs, any tags if necessary, and then click create. 
+
+Now find the new log group in the list, and click on it. The next page will show you it's details, including the ARN. Copy it for the next step.
+
+##### Create an appropriate credentials
+
+**NOTE: You should create credentials especially for use with the EventStore Cloud integration which only have access to the CloudWatch Log group, as shown here.**
```
Comment (ylorph) [inline docs/cloud/integrations/README.md]:
```suggestion
::: note 
You should create dedicated credentials for use with the EventStore Cloud integration that only have access to the CloudWatch Log group, as shown here.
:::
```

File: docs/cloud/integrations/README.md [hunk 12]
```diff
@@ -60,19 +64,231 @@ If, for some reason, the instances backing a cluster fail to provision the resou
...
+
+The EventStore Cloud backend may take up to 10 minutes to install new log integrations. 
+
+If you're making sure the integration is set up correctly it's a good idea to write a few test events to the cluster. Clusters which are seeing no activity (such as dev or test clusters) will sometimes emit no logs at all.
+
+Additionally the EventStore Cloud lacks a way to assert if the IAM credentials are valid. If no new logs appears in your AWS CloudWatch group, please double check that the credentials given to the integration are correct and that the IAM user has the appropriate permissions.
+
+### Aws CloudWatch Metrics sink
+
+**NOTE: The AwsCloudWatchMetrics sink is currently in closed beta and may change over time. To access it, please contact support.**
```
Comment (ylorph) [inline docs/cloud/integrations/README.md]:
```suggestion
::: note
The AwsCloudWatchMetrics sink is currently in closed beta and may change over time. To access it, please contact support.
:::
```

File: docs/cloud/integrations/README.md [hunk 13]
```diff
@@ -60,19 +64,231 @@ If, for some reason, the instances backing a cluster fail to provision the resou
...
+
+AWS CloudWatch is a monitoring service. The "metrics" portion of its functionality is separated into its own integration in the EventStore Cloud for ease of use.
```
Comment (ylorph) [inline docs/cloud/integrations/README.md]:
```suggestion
AWS CloudWatch is a monitoring service. The "metrics" portion of its functionality is separated into its own integration in EventStore Cloud for ease of use.
```

File: docs/cloud/integrations/README.md [hunk 14]
```diff
@@ -60,19 +64,231 @@ If, for some reason, the instances backing a cluster fail to provision the resou
...
+
+You can cause all of the metrics generated by each EventStore DB running in your clusters by creating a custom IAM user and setting up an integration as described below.
+
+*NOTE: If you're using Terraform, instructions on how to perform the steps below can be found [here](https://github.com/EventStore/terraform-provider-eventstorecloud/blob/trunk/docs/resources/integration_awscloudwatch_metrics.md#example-usage).*
```
Comment (ylorph) [inline docs/cloud/integrations/README.md]:
```suggestion
:::note 
If you're using Terraform, instructions on how to perform the steps below can be found [here](https://github.com/EventStore/terraform-provider-eventstorecloud/blob/trunk/docs/resources/integration_awscloudwatch_metrics.md#example-usage).
:::
```

File: docs/cloud/integrations/README.md [hunk 15]
```diff
@@ -60,19 +64,231 @@ If, for some reason, the instances backing a cluster fail to provision the resou
...
+
+#### Create a new AWS IAM resources
+
+**NOTE: You should create credentials especially for use with the EventStore Cloud integration which only have permission to perform the `cloudwatch:PutMetricData` action on a small set of namespaces, as shown here.**
```
Comment (ylorph) [inline docs/cloud/integrations/README.md]:
```suggestion
::: note
You should create credentials especially for use with the EventStore Cloud integration which only have permission to perform the `cloudwatch:PutMetricData` action on a small set of namespaces, as shown here.
:::
```

**Review by ylorph — CHANGES_REQUESTED** [review]

File: docs/cloud/integrations/README.md [hunk 16]
```diff
@@ -60,19 +64,247 @@ If, for some reason, the instances backing a cluster fail to provision the resou
...
+4. Under `Configuration`, by `Group Name` enter the name of the CloudWatch group you created earlier (for example, `EventStoreLogs-Production`).
+5. Under `Configuration`, by `Aws Region` enter the AWS region of the CloudWatch group you created earlier.
+6. Under `Configuration`, by `Clusters` select one or more clusters whose logs you want sent to the CloudWatch group.
+7. Under `Credentials` enter Access Key Id which was shown when you created the IAM user earlier.
+8. Under `Credentials` enter Secret Access Key which was shown when you created the IAM user earlier.
+
+
+#### Testing the integration
+
+The EventStore Cloud backend may take up to 10 minutes to install new log integrations.
```
Comment (ylorph) [inline docs/cloud/integrations/README.md]:
```suggestion
Log integration with EventStore Cloud will take a few minutes to be fully active
```
Trying not to give expectations there .

File: docs/cloud/integrations/README.md [hunk 17]
```diff
@@ -60,19 +64,247 @@ If, for some reason, the instances backing a cluster fail to provision the resou
...
+1. Under `Configuration`, by `Namespace` enter the namespace you decided on earlier (for example, `EventStoreLogs-Production`).
+2. Under `Configuration`, by `Aws Region` enter the AWS region that matches the clusters you'll be pulling from.
+3. Under `Configuration`, by `Clusters` select one or more clusters whose metrics you want sent to CloudWatch.
+4. Under `Credentials` enter Access Key Id which was shown when you created the IAM user earlier.
+5. Under `Credentials` enter Secret Access Key which was shown when you created the IAM user earlier.
+
+
+#### Testing the integration
+
+The EventStore Cloud backend may take up to 10 minutes to install new metrics integrations.
```
Comment (ylorph) [inline docs/cloud/integrations/README.md]:
```suggestion
Metrics integration with EventStore Cloud will take a few minutes to be fully activate
```
Trying to not give expectations on timing

**Review by ylorph — CHANGES_REQUESTED** [review]

**Review by alexeyzimarev — APPROVED** [review]

**Review by ylorph — APPROVED** [review]

**Pull request merged**